display improvements

edublancas opened the pull request:
## Describe your changes

## Issue number

Closes #X

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

yafimvo commented on src/sql/connection.py:
```diff
@@ -359,27 +382,53 @@ def assign_name(cls, engine):
+                if alias:
+                    display.message(f"Running query in {alias!r}")
+                else:
+                    display.message(f"Running query in {conn['url']!r}")
```
custom engines might won't have the `url` attribute (e.g. `pg.connect()`).
When we run a custom connection without an alias it will print:

![image](https://github.com/ploomber/jupysql/assets/103026689/30d383b9-d307-454f-9305-ffc7ea352f3d)

The connections are:
![image](https://github.com/ploomber/jupysql/assets/103026689/c3e20e73-e891-482c-8177-13c85a7ece9d)

edublancas replied:
I forgot about custom connections. What did we print before when listing connections?

yafimvo replied:
![image](https://github.com/ploomber/jupysql/assets/103026689/8ef9dc78-d252-48a6-bfcd-b9ceb054243f)

edublancas replied:
great catch! I realized the problem is that I wasn't escaping the HTML characters.

yafimvo reviewed (changes requested):
lgtm. added minor notes.

edublancas commented on doc/community/developer-guide.md:
````diff
@@ -23,6 +23,22 @@ Before continuing, ensure you have a working [development environment.](https://
+```
+
+```{code-cell} ipython3
+message("Some information")
````
good point. added it.

edublancas commented on doc/community/developer-guide.md:
````diff
@@ -23,6 +23,27 @@ Before continuing, ensure you have a working [development environment.](https://
+Use the `sql.display` module instead of `print` for showing feedback to the user.
+```
+
+You can use `message` (contextual information) and `message_success` (successful operations) to show feedback to the user. Here's an example:
````
https://github.com/ploomber/jupysql/issues/548

edublancas commented on src/sql/magic.py:
```diff
@@ -537,7 +538,7 @@ def _persist_dataframe(
             )
 
-        return "Persisted %s" % table_name
+        display.message_success(f"Success! Persisted {table_name} to the database.")
```
the docs say:

```
Use the `sql.display` module instead of `print` for showing feedback to the user.
```

that's enough, right?

edublancas commented on src/tests/test_run.py:
```diff
@@ -143,27 +143,35 @@ def test_sql_is_empty(mock_conns, mock_config):
     monkeypatch.setattr("sql.run._commit", mock__commit)
     monkeypatch.setattr("sql.run.handle_postgres_special", Mock())
-    monkeypatch.setattr("sql.run.interpret_rowcount", Mock())
+    monkeypatch.setattr("sql.run.display_affected_rowcount", Mock())
```
added a bunch of new tests

edublancas commented:
@yafimvo, @idomic: addressed all the comments. please give it another pass.

one important thing:

while debugging my tests I realized we had some shared state (a new test would start with existing connections from previous tests), so I added a `Connections.close_all()` method and then ensured that the pytest fixtures would clean up connections.

This led to another problem: I found out a buggy test that started failing because it was inadvertently using a connection from another test. Since the test was checking legacy behavior, I removed the test

yafimvo reviewed (approved):
lgtm.
